inlong-sort/sort-flink/sort-flink-v1.13/sort-connectors/mysql-cdc/src/main/java/org/apache/inlong/sort/cdc/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -397,6 +397,14 @@ private void closeExecutorService() {
         }
     }
 
+    private void addNewlyAddedSplits(List<MySqlSchemalessSnapshotSplit> schemaLessSnapshotSplits) {
+        int size = schemaLessSnapshotSplits.size();
+        // move the last snapshot split to the front of the remaining splits to prevent OOM
+        // caused by the excessive data of the last snapshot split.
+        remainingSplits.add(0, schemaLessSnapshotSplits.get(size - 1));
+        remainingSplits.addAll(schemaLessSnapshotSplits.subList(0, size - 1));
```

Comment:
`remainSplits` is all splits, `remainingSplits.add(0, schemaLessSnapshotSplits.get(size - 1));` will cause other table split lost

Reply:
The `remainingSplits.add(0, schemaLessSnapshotSplits.get(size - 1));` means  Inserts the specified element at the specified position in this list, it's not cause other split lost.